platform/cxxsupport/mstd_span

```diff
@@ -0,0 +1,365 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2019 ARM Limited
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef MSTD_SPAN_
+#define MSTD_SPAN_
+
+#include <mstd_iterator>
+#include <mstd_type_traits>
+
+namespace mstd {
+
+inline constexpr size_t dynamic_extent = -1;
```

Comment:
Explicitly inline variables is C++17, I think - we're using C++14. I guess GCC isn't being strict about that?

Not sure if this should then be just `constexpr` (which makes it implicitly static - is that legal when used in a template) or `extern constexpr` with an out-of-line definition. Are there any similar things in `mstd`?

Reply:
I wouldn't know about GCC. I only tested it against IAR so far, which supports C++17 so i guess that's why it works.

Reply:
IIRC, IAR doesn't even have a "dialect" switch. We are requesting C++14 from GCC and IAR.

Given where the 3 toolchains are, we're probably not far off the point we could start assuming C++17 language, if not library.

Reply:
I have tested with GCC and needed to do a few tweaks but its ok now. I just removed the inline keyword.